docs/static_analysis.md

```diff
@@ -1,33 +1,296 @@
+
+_Example:_
+```
+x.call('something');
+x.send(1 wei);
+```
+-   **Blockhash usage: Semantics maybe unclear**
+
+`blockhash` is used to access the last 256 block hashes. A miner computes the block hash by "summing up" the information in the current block mined. By summing up the information in a clever way a miner can try to influence the outcome of a transaction in the current block.
+
+_Example:_
+```
+bytes32 b = blockhash(100);
+```
+-   **Selfdestruct: Beware of caller contracts**
+
+`selfdestruct` can block calling contracts unexpectedly. Be especially careful if this contract is planned to be used by other contracts (i.e. library contracts, interactions). Selfdestruction of the callee contract can leave callers in an inoperable state.
+
+_Example:_
+```
+selfdestruct(address(0x123abc..));
+```
+
+### Category: Gas & Economy
+-   **Gas costs: Too high gas requirement of functions**
+
+If the gas requirement of a function is higher than the block gas limit, it cannot be executed. Please avoid loops in your functions or actions that modify large areas of storage
+
+_Example:_
+```
+for (uint8 proposal = 0; proposal < proposals.length; proposal++) {
+    if (proposals[proposal].voteCount > winningVoteCount) {
+        winningVoteCount = proposals[proposal].voteCount;
+        winningProposal = proposal;
+    }
+}
+```
+
+-   **This on local calls: Invocation of local functions via 'this'**
+
+Never use `this` to call functions in the same contract, it only consumes more gas than normal local calls.
+
+_Example:_
+```
+contract test {
+    
+    function callb() public {
+        address x;
+        this.b(x);
+    }
+    
+    function b(address a) public returns (bool) {}
+}
+```
+
+-   **Delete on dynamic Array: Use require/assert appropriately**
+
+The `delete` operation when applied to a dynamically sized array in Solidity generates code to delete each of the elements contained. If the array is large, this operation can surpass the block gas limit and raise an OOG exception. Also nested dynamically sized objects can produce the same results.
+
+_Example:_
+```
+contract arr {
+    uint[] users;
+    function resetState() public{
+        delete users;
+    }
+}
+```
+
+-   **For loop over dynamic array: Iterations depend on dynamic array's size**
+
+Loops that do not have a fixed number of iterations, for example, loops that depend on storage values, have to be used carefully: Due to the block gas limit, transactions can only consume a certain amount of gas. The number of iterations in a loop can grow beyond the block gas limit which can cause the complete contract to be stalled at a certain point. Additionally, using unbounded loops incurs in a lot of avoidable gas costs. Carefully test how many items at maximum you can pass to such functions to make it successful.
```

Comment:
I think these sentences should be edited:
The number of iterations in a loop can grow beyond the block gas limit which can cause the complete contract to be stalled at a certain point. Additionally, using unbounded loops incurs in a lot of avoidable gas costs.

The number of iterations in a loop can grow beyond the block gas limit which can stall the contract. 
(uhhh - does that have the same meaning? - my edit just above)

Additionally, using unbounded loops can incur avoidable gas costs.

Reply:
done

Reply:
I'm sorry, I put a typo in my edit - this is the correct sentence:
Additionally, using unbounded loops incurs a lot of avoidable gas costs.